pulser-core/pulser/register/mappable_reg.py

```diff
@@ -72,6 +72,11 @@ def build_register(self, qubits: Mapping[QubitId, int]) -> BaseRegister:
             raise ValueError(
                 "All qubits must be labeled with pre-declared qubit IDs."
             )
+        elif set(chosen_ids) != set(self.qubit_ids[: len(chosen_ids)]):
+            raise ValueError(
+                f"'qubits' should contain the {len(qubits.keys())} first "
+                "elements of the 'qubit_ids'."
+            )
```

Comment:
I think this message is a bit misleading because it makes you think you need to declare a given number of qubits (which coincidentally matches the number you are trying). Something a bit more verbose but clearer could be:
```suggestion
        elif set(chosen_ids) != set(self.qubit_ids[: len(chosen_ids)]):
            raise ValueError(
                f"To declare {len(qubits.keys())} qubits, 'qubits' should contain"
                f" the first {len(qubits.keys())} elements of the 'qubit_ids'."
            )
```